### ISSUE TITLE
Numba error in lowering function call

### ISSUE THREAD

**Issue by phansel** [numba/numba#3383]
Hi - not sure I'll be able to provide a reproducer for this error, but I'll try. Might this be an error due to multi-line function definitions/calls?

```
Traceback (most recent call last):
  File "/home/paul/.local/lib/python3.5/site-packages/numba/errors.py", line 577, in new_error_context
    yield
  File "/home/paul/.local/lib/python3.5/site-packages/numba/lowering.py", line 254, in lower_block
    self.lower_inst(inst)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/objmode.py", line 65, in lower_inst
    value = self.lower_assign(inst)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/objmode.py", line 159, in lower_assign
    return self.lower_expr(value)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/objmode.py", line 242, in lower_expr
    argvals = [self.loadvar(a.name) for a in expr.args]
  File "/home/paul/.local/lib/python3.5/site-packages/numba/objmode.py", line 242, in <listcomp>
    argvals = [self.loadvar(a.name) for a in expr.args]
  File "/home/paul/.local/lib/python3.5/site-packages/numba/objmode.py", line 520, in loadvar
    assert name in self._live_vars, name
AssertionError: data_x

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/py/lib.py", line 581, in nonJitFunction
    out,d = non_jitted_function(str1,str2,str3,str4)
  File "/home/py/lib.py", line 436, in load_data_and_map
    data_outm = other_jitted_function(str1,str2,str3,str4)
  File "/home/py/lib.py", line 324, in non_jitted_function3
    a2x2array)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/dispatcher.py", line 368, in _compile_for_args
    raise e
  File "/home/paul/.local/lib/python3.5/site-packages/numba/dispatcher.py", line 325, in _compile_for_args
    return self.compile(tuple(argtypes))
  File "/home/paul/.local/lib/python3.5/site-packages/numba/dispatcher.py", line 653, in compile
    cres = self._compiler.compile(args, return_type)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/dispatcher.py", line 83, in compile
    pipeline_class=self.pipeline_class)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 873, in compile_extra
    return pipeline.compile_extra(func)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 367, in compile_extra
    return self._compile_bytecode()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 804, in _compile_bytecode
    return self._compile_core()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 791, in _compile_core
    res = pm.run(self.status)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 253, in run
    raise patched_exception
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 245, in run
    stage()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 438, in stage_objectmode_frontend
    cres = self.frontend_looplift()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 428, in frontend_looplift
    lifted=tuple(loops), lifted_from=None)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 887, in compile_ir
    lifted_from=lifted_from)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 375, in compile_ir
    return self._compile_ir()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 811, in _compile_ir
    return self._compile_core()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 791, in _compile_core
    res = pm.run(self.status)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 253, in run
    raise patched_exception
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 245, in run
    stage()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 652, in stage_objectmode_backend
    self._backend(lowerfn, objectmode=True)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 628, in _backend
    lowered = lowerfn()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 601, in backend_object_mode
    self.flags)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/compiler.py", line 1018, in py_lowering_stage
    lower.lower()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/lowering.py", line 173, in lower
    self.lower_normal_function(self.fndesc)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/lowering.py", line 214, in lower_normal_function
    entry_block_tail = self.lower_function_body()
  File "/home/paul/.local/lib/python3.5/site-packages/numba/lowering.py", line 239, in lower_function_body
    self.lower_block(block)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/lowering.py", line 254, in lower_block
    self.lower_inst(inst)
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/errors.py", line 585, in new_error_context
    six.reraise(type(newerr), newerr, tb)
  File "/home/paul/.local/lib/python3.5/site-packages/numba/six.py", line 659, in reraise
    raise value
numba.errors.LoweringError: Failed at object (object mode frontend)
Failed at object (object mode backend)
data_x

File "lib.py", line 331:
def jitted_function(data_here, data_here2, str, str, str,
    <source elided>
    str):
    # data_here is an array of dictionaries containing numpy arrays.
    for num, d in enumerate(data_here):
    ^
[1] During: lowering "$224 = call $223(data_here, data_x, data_y, str1, vararg=None, func=$223, kws=(), args=[Var(data_here, /home/py/lib.py (331)), Var(data_x, /home/py/lib.py (334)), Var(data_y, /home/py/lib.py (335)), Var(str1, /home/py/lib.py (331))])" at /home/py/lib.py (331)
-------------------------------------------------------------------------------
This should not have happened, a problem has occurred in Numba's internals.

Please report the error message and traceback, along with a minimal reproducer
at: https://github.com/numba/numba/issues/new

If more help is needed please feel free to speak to the Numba core developers
directly at: https://gitter.im/numba/numba

Thanks in advance for your help in improving Numba!


```

**Comment by stuartarchibald** [comment]
Thanks for the report. Without a reproducer this may be hard to diagnose. There were a couple of important fixes made to liveness analysis (which is what is causing this code to fail) that were released in version 0.40.0. Are you using this version? If not, it's the first thing I'd suggest trying. Thanks.

**Comment by phansel** [comment]
@stuartarchibald, thanks for the reminder! I've been using numba 0.39.0, so I'll update and see if this error persists.

**Comment by phansel** [comment]
It looks like the fixes in 0.40.0 solved the error.

**Comment by stuartarchibald** [comment]
Great, glad it fixed it for you, thanks for confirming. Closing.
